src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -498,14 +504,19 @@ else if (qp.getType() == Leader.SNAP) {
                            throw new Exception("changes proposed in reconfig");
                        }
                     }
-                    if (!snapshotTaken) { // true for the pre v1.0 case
-                       zk.takeSnapshot();
+                    if (isPreZAB1_0) {
+                        zk.takeSnapshot();
                         self.setCurrentEpoch(newEpoch);
                     }
                     self.setZooKeeperServer(zk);
                     self.adminServer.setZooKeeperServer(zk);
                     break outerLoop;
-                case Leader.NEWLEADER: // it will be NEWLEADER in v1.0        
+                case Leader.NEWLEADER: // Getting NEWLEADER here instead of in discovery 
+                    // means this is Zab 1.0
+                    // Create updatingEpoch file and remove it after current
```

Comment:
This comment does not apply here - my guess is it is a copy paste error from your branch-3.4 patch... can we get this comment removed?